tests/resources/dirty_model.cellml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<model xmlns="http://www.cellml.org/cellml/2.0#"
+    xmlns:cellml="http://www.cellml.org/cellml/2.0#" name="dirtyModel">
+  <units name="namedEmptyUnits"/>
+  <units/>
+  <units id="nonEmptyId"/>
+  <component name="namedEmptyComponent"/>
+    <component name="nonEmptyComponent">
+        <variable name="x" units="requiredUnits"/>
+    </component>
+  <component />
```

Comment:
Formatting (to make it easier to read/review/maintain/etc.):
```xml
  <component name="nonEmptyComponent">
    <variable name="x" units="requiredUnits"/>
  </component>
  <component/>
```

Reply:
615f237